### ISSUE TITLE
Constant resend of ack'd messages on a deleted conversation

### ISSUE THREAD

**Issue by lisbakke** [layerhq/node-layer-api#20]
We're seeing a message being constantly re-sent to us. We have ack'd the receipt of it, but it is still being resent.

The message is on a deleted conversation. Could this be the cause?

get conversation by id response
```
   { id: 'object_deleted',
     code: 103,
     message: 'The referenced conversation has been deleted.',
     url: 'https://developer.layer.com/docs/platform' } 
```

Here is the message ID:

`layer:///messages/70f7d1bd-5408-450d-bc57-f0234f045191`

Here is its conversation ID:

`https://api.layer.com/apps/<layer app id>/conversations/fd93e783-2498-4deb-9b5e-3d74ff4c52a8`

**Comment by michaelkantor** [comment]
If a user is no longer a participant in a conversation, but have not yet deleted the conversation, they still see a message as "new/unread" and any attempt to send read receipts will fail (they no longer have permissions to modify these messages) and so the message continues to appear as new.

Does that sound like what your seeing?  If so: its a known problem, and not one we have any short term solutions to.  What you can do for now is test to see if the user shows up in the participants list (participants list is actually empty array if they are not a participant); and if they are not, then ignore unread messages.

michael

**Comment by lisbakke** [comment]
Let me make sure I understand -- it sounds like you're saying a participant that is no longer on a conversation can not mark the message as "read".

That's not really our problem -- the problem is we're seeing the message constantly re-sent to our webhook URL even though we call the `done` method each time.

We have two layer messages that are stuck in this re-send loop, both messages are on conversations that are deleted.

**Comment by michaelkantor** [comment]
Your seeing the same message, same message-id repeatedly being sent to you via a webhook? (please verify that the message-id is the same; that has a big impact on my understanding of this).

When you log into the developer dashboard, and view your webhooks, what status does show for your webhook; is it reporting any problems?  Is it possible that your server is taking more than 1 second to respond such that its treating this as a failed webhook and retrying?  Is there any exponential backoff in play with these repeated webhook events?

**Comment by lisbakke** [comment]
Yep, the same two message-ids are being constantly resent.

`layer:///messages/70f7d1bd-5408-450d-bc57-f0234f045191` and `layer:///messages/1ef7fe86-b5f1-47fc-9285-09adb167e98d`

EACH of these two messages are being re-sent to us at 2QPS.

The webhook is working fine -- other messages are all working correctly and the dashboard shows us as green.

**Comment by lisbakke** [comment]
I don't believe these message should be taking more than 1second to respond.

There's no backoff in the repeated webhooks and no backoff on our side.

**Comment by lisbakke** [comment]
Our code that created these messages is test code -- it creates a conversation and a message on the conversation and at the end of the test it deletes the conversation. Most of the time it runs fine, but it seems that these two messages have caused issues. It seems like a race.

On your side -- can you guys accept the `done` acknowledgement for a message that is on a conversation that no longer exists?
